R/otp-route.R

```diff
@@ -0,0 +1,287 @@
+#' the full_elevation parameter will return a nested data.frame with three columns.
+#' first and second are returned from OTP, while distance is the cumulative distance along the
+#' route and is derived from First.
+#'
+otp_plan <- function(otpcon = NA,
+           fromPlace = NA,
+           toPlace = NA,
+           mode = "CAR",
+           date_time = Sys.time(),
+           arriveBy = FALSE,
+           maxWalkDistance = 800,
+           walkReluctance = 2,
+           transferPenalty = 0,
+           minTransferTime = 0,
+           full_elevation = FALSE)
+{
+  # Check Valid Inputs
+  if(!"otpconnect" %in% class(otpcon)){
+    message("otpcon is not a valid otpconnect object")
+    stop()
+  }
+  if(class(fromPlace) != "numeric" | length(fromPlace) != 2){
+    message("fromPlace is not a valid latitude, longitude pair")
+    stop()
+  }else{
+    if(fromPlace[1] <= 90 & fromPlace[1] >= -90 &  fromPlace[2] <= 180 & fromPlace[2] >= -180){
+      fromPlace <- paste(fromPlace, collapse = ",")
+    }else{
+      message("fromPlace coordinates excced valid values +/- 90 and +/- 180 degrees")
+      stop()
+    }
+
+  }
+  if(class(toPlace) != "numeric" | length(toPlace) != 2){
+    message("toPlace is not a valid latitude, longitude pair")
+    stop()
+  }else{
+    if(toPlace[1] <= 90 & toPlace[1] >= -90 &  toPlace[2] <= 180 & toPlace[2] >= -180){
+      toPlace <- paste(toPlace, collapse = ",")
+    }else{
+      message("toPlace coordinates excced valid values +/- 90 and +/- 180 degrees")
+      stop()
+    }
+
+  }
+  if(!all(mode %in% c("TRANSIT","WALK","BICYCLE","CAR","BUS","RAIL"))){
+    message("mode is not a valid, can be a character vector of any of these values TRANSIT, WALK, BICYCLE, CAR, BUS, RAIL")
+    stop()
+  }else{
+    mode <- paste(mode, collapse = ",")
+  }
+  if(!all(class(date_time) %in% c("POSIXct","POSIXt") )){
+    message("date_time is not a valid, must be object of class POSIXct POSIXt")
+    stop()
+  }else{
+    date <- substr(as.character(date_time),1,10)
+    time <- substr(as.character(date_time),12,16)
+  }
+  if(arriveBy){
+    arriveBy <- 'true'
+  }else{
+    arriveBy <- 'false'
+  }
+
+  # Construct URL
+  routerUrl <- make_url(otpcon)
+  routerUrl <- paste0(routerUrl,"/plan")
+
+  req <- httr::GET(
+    routerUrl,
+    query = list(
+      fromPlace = fromPlace,
+      toPlace = toPlace,
+      mode = mode,
+      date = date,
+      time = time,
+      maxWalkDistance = maxWalkDistance,
+      walkReluctance = walkReluctance,
+      arriveBy = arriveBy,
+      transferPenalty = transferPenalty,
+      minTransferTime = minTransferTime
+    )
+  )
+
+  # convert response content into text
+  text <- httr::content(req, as = "text", encoding = "UTF-8")
+  # parse text to json
+  asjson <- jsonlite::fromJSON(text)
+
+  # Check for errors - if no error object, continue to process content
+  if(is.null(asjson$error$id)){
+    response <- otp_json2sf(asjson, full_elevation)
+    return(response)
```

Comment:
@mem48

Need to deal with the \'walkTime\' label bug, where it is used for cycle time and drive time too. See line 162 etc. in my original function (in R scripts folder).

Reply:
I would not change this, this is an OTP bug and should be fixed in the OTP, if we start making these changes it will become a nightmare supporting different versions of OTP and moves the package away from being a simple bridge between OTP and R

Reply:
Agreed. I will submit bug report on this to OTP (has been around a long time).